File: wordlist.py
import os
import random
import readline

# Define constants for color formatting
RED_COLOR = "\033[0;31m"
GREEN_COLOR = "\033[0;32m"
YELLOW_COLOR = "\033[0;33m"
WHITE_COLOR = "\033[0;37m"
BOLD_TEXT = "\033[5;1m"
BACKGROUND_RED = "\033[41m"
RESET_FORMAT = "\033[0m"

# Define password formats
PASSWORD_FORMATS = {
    '1': "0123456789",
    '2': "ABCDEFGHIJKLMNOPQRSTUVWXYZ",
    '3': "abcdefghijklmnopqrstuvwxyz",
    '4': "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz",
    '5': "@#$&(_-=%*':/!?+÷;`]}><{[^¡¿~™®©¢¥€£¶§×`",
    '6': "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789",
    '7': "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789@#$&(_-=%*':/!?+÷;`]}><{[^¡¿~™®©¢¥€£¶§×`"
}

def main():
    print(f"{RED_COLOR}{BOLD_TEXT}")
    os.system("cat logo/logo.txt")
    print(f"{WHITE_COLOR}")

    password_format = get_password_format()
    num_passwords, password_length = get_password_info()
    output_file = get_file_name()

    generated_passwords = generate_passwords(password_format, num_passwords, password_length, output_file)
    save_passwords(output_file, generated_passwords)

def get_password_format():
    while True:
        select = input(f"{GREEN_COLOR}{BOLD_TEXT}\n╭── Choose the password format\n│\n"
                       f"├─ [1] {RED_COLOR}1234... {GREEN_COLOR}\n"
                       f"├─ [2] {RED_COLOR}ABCD... {GREEN_COLOR}\n"
                       f"├─ [3] {RED_COLOR}abcd... {GREEN_COLOR}\n"
                       f"├─ [4] {RED_COLOR}ABcd... {GREEN_COLOR}\n"
                       f"├─ [5] {RED_COLOR}@#$&... {GREEN_COLOR}\n"
                       f"├─ [6] {RED_COLOR}Ac23... {GREEN_COLOR}\n"
                       f"├─ [7] {RED_COLOR}Ab3&... {GREEN_COLOR}\n"
                       f"│\n╰────▶{YELLOW_COLOR} ")
        if select in PASSWORD_FORMATS:
            return select

def get_password_info():
    while True:
        try:
            num_passwords = int(input(f"{GREEN_COLOR}{BOLD_TEXT}│\n│\n├── How many passwords do you need ? [Ex:1000]\n│\n│\n"
                                    f"╰────▶{YELLOW_COLOR}{BOLD_TEXT} "))
            password_length = int(input(f"{GREEN_COLOR}{BOLD_TEXT}│\n│\n├── How long is password ? [Ex:8]\n│\n│\n"
                                       f"╰────▶{YELLOW_COLOR}{BOLD_TEXT} "))
            return num_passwords, password_length
        except ValueError:
            print(f"{RED_COLOR}{BOLD_TEXT}│\n│\n├── {WHITE_COLOR}{BACKGROUND_RED}Enter a number!{RESET_FORMAT}")

def get_file_name():
    while True:
        output_file = input(f"{GREEN_COLOR}{BOLD_TEXT}│\n│\n├── Enter the filename to save passwords [Ex: pass.txt]\n│\n│\n"
                            f"╰────▶{YELLOW_COLOR}{BOLD_TEXT}")
        if output_file == "":
            output_file = "pass.txt"
        if output_file[-4:] != ".txt":
            output_file += ".txt"
        return output_file

def generate_passwords(password_format, num_passwords, password_length, output_file):
    characters = PASSWORD_FORMATS[password_format]
    passwords = set()

    count = 0
    while count < num_passwords:
        password = "".join(random.choice(characters) for _ in range(password_length))
        if password not in passwords:
            passwords.add(password)
            print(f"{GREEN_COLOR}{BOLD_TEXT}│\n├── {YELLOW_COLOR}{password}")
            count += 1

    print(f"{GREEN_COLOR}│\n╰────⎙ Saved as {output_file}")
    return passwords

def save_passwords(output_file, passwords):
    with open(output_file, "w") as file:
        file.writelines("\n".join(passwords))

if __name__ == "__main__":
    try:
        main()
    except KeyboardInterrupt:
        exit(f"{RED_COLOR}{BOLD_TEXT}│\n│\n│\n├── Sudden stop!\n│\n╰── Exit...{WHITE_COLOR}")

File: wordlist.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <time.h>

#define RED_COLOR   "\033[0;31m"
#define GREEN_COLOR "\033[0;32m"
#define YELLOW_COLOR "\033[0;33m"
#define WHITE_COLOR "\033[0;37m"
#define BOLD_TEXT   "\033[5;1m"
#define BACKGROUND_RED "\033[41m"
#define RESET_FORMAT "\033[0m"

#define PASSWORD_FORMATS  "0123456789", "ABCDEFGHIJKLMNOPQRSTUVWXYZ", "abcdefghijklmnopqrstuvwxyz", "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz", "@#$&(_-=%*':/!?+÷;`]}><{[^¡¿~™®©¢¥€£¶§×`", "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789", "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789@#$&(_-=%*':/!?+÷;`]}><{[^¡¿~™®©¢¥€£¶§×`"

void get_password_format(char* format);
void get_password_info(int* num_passwords, int* password_length);
void get_file_name(char* output_file);
void generate_passwords(char* password_format, int num_passwords, int password_length, char* output_file);

int main() {
    char format[2];
    get_password_format(format);
    int num_passwords, password_length;
    get_password_info(&num_passwords, &password_length);
    char output_file[100];
    get_file_name(output_file);

    generate_passwords(format, num_passwords, password_length, output_file);
    return 0;
}

void get_password_format(char* format) {
    while (1) {
        printf("%s%s\n", GREEN_COLOR, BOLD_TEXT);
        printf("Enter the password format:\n");
        printf("[1] 1234...\n");
        printf("[2] ABCD...\n");
        printf("[3] abcd...\n");
        printf("[4] ABcd...\n");
        printf("[5] @#$&...\n");
        printf("[6] Ac23...\n");
        printf("[7] Ab3&...\n");
        printf("%s", YELLOW_COLOR);

        char select[2];
        fgets(select, sizeof(select), stdin);
        select[strlen(select) - 1] = '\0';

        if (strcmp(select, "1") == 0 || strcmp(select, "2") == 0 || strcmp(select, "3") == 0 || strcmp(select, "4") == 0 || strcmp(select, "5") == 0 || strcmp(select, "6") == 0 || strcmp(select, "7") == 0) {
            strncpy(format, select, 2);
            break;
        }
    }
}

void get_password_info(int* num_passwords, int* password_length) {
    while (1) {
        printf("%s%s\n", GREEN_COLOR, BOLD_TEXT);
        printf("How many passwords do you need? [Ex:1000]:\n");
        printf("%s", YELLOW_COLOR);

        char input[10];
        fgets(input, sizeof(input), stdin);
        *num_passwords = atoi(input);

        if (*num_passwords > 0) {
            break;
        }
    }

    while (1) {
        printf("%s%s\n", GREEN_COLOR, BOLD_TEXT);
        printf("How long is the password? [Ex:8]:\n");
        printf("%s", YELLOW_COLOR);

        char input[10];
        fgets(input, sizeof(input), stdin);
        *password_length = atoi(input);

        if (*password_length > 0) {
            break;
        }
    }
}

void get_file_name(char* output_file) {
    while (1) {
        printf("%s%s\n", GREEN_COLOR, BOLD_TEXT);
        printf("Enter the filename to save passwords [Ex: pass.txt]:\n");
        printf("%s", YELLOW_COLOR);

        fgets(output_file, 100, stdin);
        output_file[strlen(output_file) - 1] = '\0';

        if (strlen(output_file) == 0) {
            strcpy(output_file, "pass.txt");
        }

        if (output_file[strlen(output_file) - 4] != '.') {
            strcat(output_file, ".txt");
        }

        break;
    }
}

void generate_passwords(char* password_format, int num_passwords, int password_length, char* output_file) {
    const char* characters = PASSWORD_FORMATS[*password_format - '1'];
    char passwords[num_passwords][password_length + 1];
    int count = 0;

    while (count < num_passwords) {
        char password[password_length + 1];
        password[password_length] = '\0';

        for (int i = 0; i < password_length; i++) {
            int index = rand() % strlen(characters);
            password[i] = characters[index];
        }

        int is_duplicate = 0;

        for (int i = 0; i < count; i++) {
            if (strcmp(passwords[i], password) == 0) {
                is_duplicate = 1;
                break;
            }
        }

        if (!is_duplicate) {
            strcpy(passwords[count], password);
            printf("%s%s\n", GREEN_COLOR, BOLD_TEXT);
            printf("%s", YELLOW_COLOR);
            printf("Password: %s\n", password);
            count++;
        }
    }

    FILE* file = fopen(output_file, "w");

    for (int i = 0; i < num_passwords; i++) {
        fprintf(file, "%s\n", passwords[i]);
    }

    fclose(file);
    printf("%s%s\n", GREEN_COLOR, BOLD_TEXT);
    printf("Saved as %s\n", output_file);
}
